cmd/soroban-rpc/internal/db/ledgerentry.go

```diff
@@ -148,6 +197,20 @@ func (l *ledgerEntryReadTx) GetLedgerEntry(key xdr.LedgerKey) (bool, xdr.LedgerE
 	if err = xdr.SafeUnmarshal([]byte(ledgerEntryBin), &result); err != nil {
 		return false, xdr.LedgerEntry{}, err
 	}
+
+	// Disallow access to entries that have expired. Expiration excludes the
+	// "current" ledger, which we are building.
+	// TODO: Support allowing access, but recording for simulateTransaction
+	if expirationLedgerSeq, ok := result.Data.ExpirationLedgerSeq(); ok {
+		latestLedger, err := l.GetLatestLedgerSequence()
+		if err != nil {
+			return false, xdr.LedgerEntry{}, err
+		}
+		if expirationLedgerSeq <= xdr.Uint32(latestLedger) {
```

Comment:
@dmkozh how does the core treat expiration ledger ? is it inclusive or exclusive ?
i.e. should it be like the above or just `<` ?
does it mean "the last ledger at which the data is available" or "the first ledger where the data is no longer availble" ?

Reply:
it's the former, i.e. the entry is still available on expiration ledger. So this should be `<`.

Reply:
`latestLedger` here refers to the latest closed ledger. So this will only hide the entry once the expirationLedgerSeq ledger number is closed.